### PR TITLE
Fix compilation issues on MSVC 32 bit

### PR DESCRIPTION
**Pull request by jiayuehua**
…e on x64-windows

https://github.com/microsoft/vcpkg/pull/31921

### CODE REVIEW

**Comment by upsj** [comment]
Great catch, we are not really testing the MSVC MPI compilation ATM.

File: core/base/mtx_io.cpp [hunk 1]
```diff
@@ -845,7 +845,7 @@ matrix_data<ValueType, IndexType> read_binary_convert(std::istream& is,
         throw GKO_STREAM_ERROR(
             "cannot read into this format, would assign complex to real");
     }
-    matrix_data<ValueType, IndexType> result(gko::dim<2>{num_rows, num_cols});
+    matrix_data<ValueType, IndexType> result(gko::dim<2>{static_cast<size_type>(num_rows), static_cast<size_type>(num_cols)});
```
Comment (upsj) [inline core/base/mtx_io.cpp:848]: I guess you compiled for x86 (32 bit) here?

Comment (jiayuehua) [reply core/base/mtx_io.cpp:848]: right

**Comment by yhmtsai** [comment]
format!

**Review by yhmtsai — APPROVED** [review]
LGTM!

**Review by tcojean — APPROVED** [review]

**Pull request merged**